packages/aws-cdk/lib/api/hotswap/stepfunctions-state-machines.ts

```diff
@@ -0,0 +1,72 @@
+import { ISDK } from '../aws-auth';
+import { ChangeHotswapImpact, ChangeHotswapResult, HotswapOperation, HotswappableResourceChange, establishHotswappableResourceName } from './common';
+import { EvaluateCloudFormationTemplate } from './evaluate-cloudformation-template';
+
+export async function isHotswappableStateMachineChange(
+  logicalId: string, change: HotswappableResourceChange, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+): Promise<ChangeHotswapResult> {
+  const stateMachineDefinitionChange = await isStateMachineDefinitionOnlyChange(change, evaluateCfnTemplate);
+
+  if ((stateMachineDefinitionChange === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT) ||
+      (stateMachineDefinitionChange === ChangeHotswapImpact.IRRELEVANT)) {
```

Comment:
What's with the redundant parentheses here?

```suggestion
  if (stateMachineDefinitionChange === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT ||
      stateMachineDefinitionChange === ChangeHotswapImpact.IRRELEVANT) {
```